src/rules/a11y-use-accessible-tooltip.js

```diff
@@ -0,0 +1,153 @@
+'use strict'
+const url = require('../url')
+const {getJSXOpeningElementAttribute} = require('../utils/get-jsx-opening-element-attribute')
+const {getJSXOpeningElementName} = require('../utils/get-jsx-opening-element-name')
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'recommends the use of @primer/react Tooltip component',
+      category: 'Best Practices',
+      recommended: true,
+      url: url(module),
+    },
+    fixable: true,
+    schema: [],
+    messages: {
+      useAccessibleTooltip: 'Please use @primer/react Tooltip component that has accessibility improvements',
+      useTextProp: 'Please use the text prop instead of aria-label',
+      noDelayRemoved: 'noDelay prop is removed. Tooltip now has no delay by default',
+      wrapRemoved: 'wrap prop is removed. Tooltip now wraps by default',
+      alignRemoved: 'align prop is removed. Please use the direction prop instead.',
+    },
+  },
+  create(context) {
+    return {
+      ImportDeclaration(node) {
+        if (node.source.value !== '@primer/react/deprecated') {
+          return
+        }
+        const hasTooltip = node.specifiers.some(
+          specifier => specifier.imported && specifier.imported.name === 'Tooltip',
+        )
+
+        if (!hasTooltip) {
+          return
+        }
+
+        const hasOtherImports = node.specifiers.length > 1
+
+        const sourceCode = context.getSourceCode()
+        // Checking to see if there is an existing root (@primer/react) import
+        const rootImport = sourceCode.ast.body.filter(statement => {
+          return statement.type === 'ImportDeclaration' && statement.source.value === '@primer/react'
+        })
+
+        const tooltipSpecifier = node.specifiers.find(
+          specifier => specifier.imported && specifier.imported.name === 'Tooltip',
+        )
+
+        const hasRootImport = rootImport.length >= 1
+
+        context.report({
+          node,
+          messageId: 'useAccessibleTooltip',
+          fix(fixer) {
+            const fixes = []
+            if (!hasOtherImports) {
+              // If Tooltip is the only import and no existing @primer/react import, replace the whole import statement
+              if (!hasRootImport) fixes.push(fixer.replaceText(node.source, `'@primer/react'`))
+              if (hasRootImport) {
+                //  remove the entire import statement
+                fixes.push(fixer.remove(node))
+                // find the last specifier in the existing @primer/react import and insert Tooltip after that
+                const rootImport = sourceCode.ast.body.find(statement => {
+                  return statement.type === 'ImportDeclaration' && statement.source.value === '@primer/react'
+                })
```

Comment:
Wondering if there's a reason we can't reuse the `rootImport` from line 43 here?

Reply:
We can re-use! I missed that, due to using the wrong method (filter vs find.) I updated the code 🤗 Thanks for catching that.